File: app/src/main/java/app/simple/buyer/util/views/MultiCellObject.java
package app.simple.buyer.util.views;

import android.view.View;

import rx.functions.Func1;

/**
 * Created by Zakharovi on 12.01.2018.
 */

public class MultiCellObject <T>{
    private T object;
    private int resourceId;
    private Func1<View, BindHolder<T>> onCreateHolder;

    public MultiCellObject(T object, Func1<View, BindHolder<T>> onCreateHolder, int resourceId) {
        this.object = object;
        this.resourceId = resourceId;
        this.onCreateHolder = onCreateHolder;
    }

    public T getObject() {
        return object;
    }

    public int getResource() {
        return resourceId;
    }

    public Func1<View, BindHolder<T>> getOnCreateHolder() {
        return onCreateHolder;
    }
}

File: app/src/main/java/app/simple/buyer/util/views/MultiCellObject.kt
package app.simple.buyer.util.views

import android.view.View
import io.reactivex.functions.Function

/**
 * Created by Zakharovi on 12.01.2018.
 */

class MultiCellObject<T>(val obj: T, val onCreateHolder: Function<View, BindHolder<T>>, val resource: Int)

